syntax = "proto3";
package files;

service Files {
    rpc copyFileToContainer(File) returns (CopyResponse) {}
    rpc getFile(FileDetails) returns (stream FilesResponse) {}
    rpc getDirectory(FolderDetails) returns (DirectoryStrcuture) {}
}


message File {
    string file_name = 1;
    string path = 2;
    string container_name = 3;
    string file_content = 4;
}

message FileDetails {
    string container_name = 1;
    string file_name = 2;
    string path = 3;
}

message FolderDetails {
    string container_name = 1;
    string folder_name = 2;
    string path = 3;
}

message CopyResponse {
    bool success = 1;
}

message FilesResponse {
    string line = 1;
}

message DirectoryStrcuture {
    string yaml_rep = 1;
}
